src/hotspot/share/opto/vectorization.hpp

```diff
@@ -27,10 +27,27 @@
 
 #include "opto/node.hpp"
 #include "opto/loopnode.hpp"
+#include "opto/traceAutoVectorizationTag.hpp"
 
 // Code in this file and the vectorization.cpp contains shared logics and
 // utilities for C2's loop auto-vectorization.
 
+#ifndef PRODUCT
+// Access to TraceAutoVectorization tags
+class VTrace : public StackObj {
+private:
+  const CHeapBitMap &_trace_tags;
+
+public:
+  VTrace() : _trace_tags(Compile::current()->directive()->traceautovectorization_tags()) {}
+  NONCOPYABLE(VTrace);
+
+  bool is_trace(TraceAutoVectorizationTag tag) const {
+    return _trace_tags.at(tag);
+  }
+};
+#endif
+
```

Comment:
The idea is that this is going to be a "component" of VLoop, once I do the bigger refactoring.